src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -351,6 +353,32 @@ public static boolean isCancelled(BuildStatus status) {
     return status != null && status.getCancelled() != null && Boolean.TRUE.equals(status.getCancelled());
   }
 
+  /**
+   * Checks to see if there's a registry name already provided in the image name
+   *
+   * Code influenced by io.fabric8.kubernetes.pipeline.devops.ApplyStepExecution, which was 
+   * influenced from <a href="https://github.com/rhuss/docker-maven-plugin/blob/master/src/main/java/org/jolokia/docker/maven/util/ImageName.java">docker-maven-plugin</a>
+   * @param imageName
+   * @return true if the image name contains a registry
+   */
+  public static boolean hasRegistry(String imageName) {
+      if (imageName == null) {
+          return false;
+      }
+      Pattern tagPattern = Pattern.compile("^(.+?)(?::([^:/]+))?$");
+      Matcher matcher = tagPattern.matcher(imageName);
+      if (!matcher.matches()) {
+          // if here then imageName is not a proper image name ([registry/][repo][:port]");
+          return false;
+      }
+
+      String rest = matcher.group(1);
+      String[] parts = rest.split("\\s*/\\s*");
+      String part = parts[0];
+
+      return part.contains(".") || part.contains(":");
+  }
```

Comment:
i'm not sure about the validity of this check...

valid docker image refs:

hostname/repo  (no colons, no dots, single slash) - refers to the default "library" org (e.g. docker.io/centos)

that's a dockerimage value that has a registry but which this check would not recognize as having a registry. (hostname need not have dots in it, since in theory a DNS search path could be appending the domain).

I think the better validation to do is "is this a valid imagestreamtag" which means:
[optional_namespace]/imagestreamname:tag

so (.*/)(.*:.*)  (with appropriate escaping and "dot" is probably too loose.  but we should have this regex somewhere in the origin code already)